### PR TITLE
Feature/improve queries

### PR DESCRIPTION
**Pull request by manuelroth**
- Updated all layer with new imposm mapping and added international names to every label layer
- Moved classification out of layers into sql functions
- Improved mapping of some tables with missing types
- Removed typebased generalizations in imposm mapping
- Removed unused table amenities
- Added new layer marine to source project
- Changed all sql queries to uppercase style

### CODE REVIEW

File: open-streets.tm2source/data.yml [hunk 1]
```diff
@@ -797,33 +1061,56 @@ Layer:
...
+          type, type as class 
+          FROM osm_water_lines
+          WHERE type in ('river', 'canal')
+            AND z(!scale_denominator!) >= 8
+            AND linelabel(z(!scale_denominator!), name, geometry)
+            AND geometry && !bbox!
+          UNION ALL
+          SELECT geometry, osm_id,
+          name,
+          coalesce(name, name_de) as name_de,
```
Comment (lukasmartinelli) [inline open-streets.tm2source/data.yml]: Do we need the `COALESCE` calls in the `FROM ` clause when we do `COALESCE` in the `SELECT` as well?

File: src/prepare-open-streets/functions.sql [hunk 2]
```diff
@@ -2,14 +2,18 @@ CREATE OR REPLACE FUNCTION classify_road(type VARCHAR) RETURNS VARCHAR
 AS $$
 BEGIN
     RETURN CASE
-        WHEN type IN ('motorway', 'motorway_link', 'driveway') THEN 'highway'
+        WHEN type IN ('motorway') THEN 'motorway'
```
Comment (lukasmartinelli) [inline src/prepare-open-streets/functions.sql]: Is this already the generated version :grin: ?

File: open-streets.tm2source/data.yml [hunk 3]
```diff
@@ -565,92 +708,192 @@ Layer:
...
+          coalesce(name, name_en) AS name_en,
+          coalesce(name, name_es) AS name_es,
+          coalesce(name, name_fr) AS name_fr,
+          coalesce(name, name_de) AS name_de,
+          coalesce(name, name_ru) AS name_ru,
+          coalesce(name, name_zh) AS name_zh,
+          area 
+          FROM osm_water_polygons
+          WHERE area < 1000000 AND area >= 500000
+            AND z(!scale_denominator!) >= 15
```
Comment (lukasmartinelli) [inline open-streets.tm2source/data.yml]: We can remove queries that are for a zoom level bigger than 14.

File: open-streets.tm2source/data.yml [hunk 4]
```diff
@@ -420,55 +439,140 @@ Layer:
...
+        (
+          SELECT geometry, name, 
+          coalesce(name, name_en) AS name_en,
+          coalesce(name, name_es) AS name_es,
+          coalesce(name, name_fr) AS name_fr,
+          coalesce(name, name_de) AS name_de,
+          coalesce(name, name_ru) AS name_ru,
+          coalesce(name, name_zh) AS name_zh,
+          GeometryType(geometry) as placement,
+          rank_marine(type) as labelrank
```
Comment (lukasmartinelli) [inline open-streets.tm2source/data.yml]: Cool that we can solve the ranking on marine by simply checking the types.

File: open-streets.tm2source/data.yml (hunk 1 above)
Comment (manuelroth) [inline open-streets.tm2source/data.yml]: You are totally right. I will remove them.

File: open-streets.tm2source/data.yml (hunk 4 above)
Comment (manuelroth) [inline open-streets.tm2source/data.yml]: Would be nice, if it was that easy. It is just a workaround for now. It works for the oceans, but not for seas and bays(the labelranks varys between 2-5)

File: src/prepare-open-streets/functions.sql (hunk 2 above)
Comment (manuelroth) [inline src/prepare-open-streets/functions.sql]: Not yet, but I added the additional mappings to the classify.yml file in the generate-sql branch. We can merge this tomorrow.

**Comment by manuelroth** [comment]
I really enjoy these graphs. As the source project an mapping get more complex. It is nice to have an overview over what is mapped to what.

**Comment by lukasmartinelli** [comment]
Improved some additional queries.

> Not yet, but I added the additional mappings to the classify.yml file in the generate-sql branch. We can merge this tomorrow.

Yep, let's do this later.

> I really enjoy these graphs. As the source project an mapping get more complex. It is nice to have an 
overview over what is mapped to what.

It has gotten out of control with all these layers and tables :grinning:  Difficult to fit it all into the head.

**Comment by manuelroth** [comment]
> It has gotten out of control with all these layers and tables :grinning: Difficult to fit it all into the head.
Yes, definitely. Thanks for the improvements.

**Comment by lukasmartinelli** [comment]
Replaced alot of `UNION` queries with `WHERE` statements. We only need `UNION` if we query different tables otherwise we can just specify big `OR` queries.

I hope I got it sommentically the same as before.

**Pull request merged**